python/libsettings.pyx

```diff
@@ -148,21 +148,23 @@ cdef class Settings:
     def destroy(self):
         settings_destroy(&self.ctx)
 
-    def write(self, section, name, value):
+    def write(self, section, name, value, encoding='ascii'):
+        if not isinstance(value, str):
+            value = str(value).encode(encoding)
         return settings_write_str(self.ctx,
-                                  bytes(section),
-                                  bytes(name),
-                                  bytes(str(value)))
+                                  bytearray(section, encoding),
+                                  bytearray(name, encoding),
+                                  bytearray(value, encoding))
 
-    def read(self, section, name):
+    def read(self, section, name, encoding='ascii'):
         cdef char value[SETTINGS_BUFLEN]
         ret = settings_read_str(self.ctx,
-                                bytes(section),
-                                bytes(name),
+                                bytearray(section, encoding),
+                                bytearray(name, encoding),
                                 value,
                                 sizeof(value))
         if (ret == 0):
-            return str(value)
+            return value
```

Comment:
I'm not 100% sure what would be the neatest method but now it seems that all inputs of read/read_all/write are strings but outputs are bytes/bytearrays. Wouldn't it be more symmetric if the outputs were strings, too? I.e.,
`return value.decode(encoding)` (and the same for `read_all`)? This way the application wouldn't need to do any decoding in the default uses but an extra encoding parameter (with a default value) would be needed in `read_all`.

Reply:
added decoding